3-0-spring-framework/3-3-0-enable-string-trimming/src/test/java/com/bobocode/TrimmingPostProcessorTest.java

```diff
@@ -0,0 +1,254 @@
+package com.bobocode;
+
+import static java.util.stream.Collectors.toList;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.contains;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.bobocode.config.TrimmingEnabledConfig;
+import com.bobocode.config.TrimmingNotEnabledConfig;
+import com.bobocode.service.TrimmedService;
+import java.lang.annotation.Annotation;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.lang.reflect.Method;
+import java.util.Arrays;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+import java.util.stream.Stream;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.MethodOrderer;
+import org.junit.jupiter.api.Nested;
+import org.junit.jupiter.api.Order;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestMethodOrder;
+import org.reflections.Reflections;
+import org.reflections.scanners.SubTypesScanner;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.test.context.junit.jupiter.SpringJUnitConfig;
+
+
+@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
+@SpringJUnitConfig(classes = TrimmingEnabledConfig.class)
```

Comment:
@Stanislav-Zabramnyi simple Reflection-based tests **should not be apart of Spring Context test.** Here's why, I made one mistake that broke a Spring Context, and all my tests are now failing...

<img width="1440" alt="Screenshot 2022-09-26 at 15 21 50" src="https://user-images.githubusercontent.com/10629458/192275355-83baafd0-056d-43d8-810a-b3d1348cb59f.png">

Reply:
I am not fully understand what do yo mean. Could you provide more details?